crowdkit/aggregation/classification/kos.py

```diff
@@ -0,0 +1,119 @@
+__all__ = [
+    'KOS'
+]
+
+import attr
+import numpy as np
+import pandas as pd
+
+from ..base import BaseClassificationAggregator
+
+_MAX = np.float_power(10, 300)
+
+
+@attr.s
+class KOS(BaseClassificationAggregator):
+    r"""Karger-Oh-Shah aggregation model.
+
+    Iterative algorithm that calculates the log-likelihood of the task being positive while modeling
+    the reliabilities of the workers.
+
+    Let $A_{ij}$ be a matrix of answers of worker $j$ on task $i$.
+    $A_{ij} = 0$ if worker $j$ didn't answer the task $i$, otherwise $|A_{ij}| = 1$.
+    The algorithm operates on real-valued task messages $x_{i \rightarrow j}$  and
+    worker messages $y_{j \rightarrow i}$. A task message $x_{i \rightarrow j}$ represents
+    the log-likelihood of task $i$ being a positive task, and a worker message $y_{j \rightarrow i}$ represents
+    how reliable worker $j$ is.
+
+    On iteration $k$ the values are updated as follows:
+    $$
+    x_{i \rightarrow j}^{(k)} = \sum_{j^{'} \in \partial i \backslash j} A_{ij^{'}} y_{j^{'} \rightarrow i}^{(k-1)} \\
+    y_{j \rightarrow i}^{(k)} = \sum_{i^{'} \in \partial j \backslash i} A_{i^{'}j} x_{i^{'} \rightarrow j}^{(k-1)}
+    $$
+
+    Karger, David R., Sewoong Oh, and Devavrat Shah. Budget-optimal task allocation for reliable crowdsourcing systems.
+    Operations Research 62.1 (2014): 1-24.
+
+    <https://arxiv.org/abs/1110.3564>
+
+    Args:
+        n_iter: The number of iterations.
+
+    Examples:
+        >>> from crowdkit.aggregation import KOS
+        >>> from crowdkit.datasets import load_dataset
+        >>> df, gt = load_dataset('relevance-2')
+        >>> ds = KOS(10)
+        >>> result = ds.fit_predict(df)
+
+    Attributes:
+        labels_ (Optional[pd.Series]): Tasks' labels.
+            A pandas.Series indexed by `task` such that `labels.loc[task]`
+            is the tasks's most likely true label.
+
+    """
+
+    n_iter: int = attr.ib(default=100)
+    random_state: int = attr.ib(default=0)
+
```

Comment:
Please define all the attributes here, including the ones like `labels_` that we write in `fit()` and other methods. `MajorityVote` might be a good example.